addons/delivery/models/product_template.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from openerp import api, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    @api.multi
+    def write(self, vals):
+        #TODO FixMe, onchange of template list_price change the fixed price of the delivery method and update price rules based on fixed price.
```

Comment:
I suppose the fixed_price field is being updated thanks to its inverse function... However, I don't understand why you need to call create_price_rules here since it is already called in the write function of delivery.carrier ?